packages/client/src/components/form/FormFieldGenerator.tsx

```diff
@@ -199,440 +199,465 @@ type GeneratedInputFieldProps = {
-      >
-        {fieldDefinition.label}
-      </Link>
-    )
-  }
-
-  if (
-    fieldDefinition.type === LOCATION_SEARCH_INPUT &&
-    fieldDefinition.locationList
-  ) {
-    const selectedLocation = fieldDefinition.locationList.find(
-      (location) => location.id === value
-    )
-
-    return (
-      <InputField {...inputFieldProps}>
-        <LocationSearchFormField
-          buttonLabel={intl.formatMessage(buttonMessages.search)}
-          {...inputProps}
-          selectedLocation={selectedLocation}
-          locationList={fieldDefinition.locationList}
-          searchHandler={(item) => {
-            onSetFieldValue(fieldDefinition.name, item.id)
-            if (fieldDefinition.dispatchOptions) {
-              dynamicDispatch(fieldDefinition.dispatchOptions.action, {
-                [fieldDefinition.dispatchOptions.payloadKey]: item.id
-              })
-            }
-          }}
+          maxLength={(fieldDefinition as Ii18nTextFormField).maxLength}
+          isDisabled={disabled}
         />
       </InputField>
     )
-  }
-
-  if (fieldDefinition.type === FETCH_BUTTON) {
-    return (
-      <FetchButtonField
-        id={fieldDefinition.name}
-        queryData={fieldDefinition.queryData}
-        modalTitle={fieldDefinition.modalTitle}
-        label={fieldDefinition.label}
-        successTitle={fieldDefinition.successTitle}
-        errorTitle={fieldDefinition.errorTitle}
-        onFetch={fieldDefinition.onFetch}
-        isDisabled={disabled}
-      />
-    )
-  }
-
-  if (fieldDefinition.type === NID_VERIFICATION_BUTTON) {
-    return (
-      <InputField {...inputFieldProps}>
-        <VerificationButton
-          id={fieldDefinition.name}
-          onClick={fieldDefinition.onClick}
-          labelForVerified={fieldDefinition.labelForVerified}
-          labelForUnverified={fieldDefinition.labelForUnverified}
-          labelForOffline={fieldDefinition.labelForOffline}
-          status={!isOnline ? 'offline' : value ? 'verified' : 'unverified'}
-        />
-      </InputField>
+  },
+
+  // This is a hack to workaround slow renders of Selects.
+  // A proper solution would not pass new props to this component everytime,
+  // rather they should pass the variable / function references.
+  // This may be achieved by useMemo / useCallback'ing the props before passing.
+
+  // If the function returns false, props are not equal and it will re-render
+  // If function returns true, they are equal and no rerender happens
+  (prevProps, nextProps) =>
+    prevProps.fieldDefinition.type === 'SELECT_WITH_OPTIONS' &&
+    nextProps.fieldDefinition.type === 'SELECT_WITH_OPTIONS' &&
+    prevProps.value === nextProps.value &&
+    prevProps.touched === nextProps.touched &&
+    prevProps.error === nextProps.error &&
+    prevProps.disabled === nextProps.disabled &&
+    isEqual(
+      prevProps.fieldDefinition.options,
+      nextProps.fieldDefinition.options
     )
-  }
+)
```

Comment:
It's an unfortunate diff as whitespace has changed, but these are the added lines + the React.memo at the top of this component.